gp/hl_api.py

```diff
@@ -1,18 +1,21 @@
 import numpy as np
+from typing import Optional, Callable
+from .abstract_individual import AbstractIndividual
+from .abstract_population import AbstractPopulation
+from .ea import MuPlusLambda
 
 
 def evolve(
-    pop,
-    objective,
-    ea,
-    max_generations,
-    min_fitness,
-    print_progress=False,
-    *,
-    callback=None,
-    label=None,
-    n_processes=1,
-):
+    pop: AbstractPopulation,
+    objective: Callable[[AbstractIndividual], AbstractIndividual],
+    ea: MuPlusLambda,
+    max_generations: int,
+    min_fitness: float,
+    print_progress: Optional[bool] = False,
+    callback: Optional[Callable[[AbstractPopulation], None]] = None,
```

Comment:
callback just receives the population

Reply:
yes, the `None` defines the return type.